packages/node-cli/src/services/settings/CLISettings.js

```diff
@@ -4,11 +4,26 @@
 */
 'use strict';
 
+const DEFAULT_CLI_SETTINGS = {
+	useProxy: false,
+	proxyUrl: '',
+	isJavaVersionValid: false,
+}
+
 module.exports = class CLISettings {
+	// should not be directly used, use static fromJson(json) method
 	constructor(options) {
-		this._useProxy = options.useProxy;
```

Comment:
I think you should enforce that no one use this constructor then...
I'm looking at solutions that doesn't convince myself. We could look at it tomorrow.

https://stackoverflow.com/questions/21667149/how-to-define-private-constructors-in-javascript